### PR TITLE
Add a 3D particles demo

### PR DESCRIPTION
**Pull request by Calinou**
Follow-up to https://github.com/godotengine/godot-demo-projects/pull/756.

This demo showcases various CPUParticles3D and GPUParticles3D features.

## Preview

image

### CODE REVIEW

**Comment by Calinou** [comment]
Rebased and tested again, it works as expected.

**Comment by Lippanon** [comment]
Adding more sections (just 1 more is enough, making it 6) to the TubeTrailMesh 'breaks' the particles in the GPUParticlesTrails.
section
Probably a bug

**Comment by Calinou** [comment]
@Lippanon This is an engine bug: https://github.com/godotengine/godot/issues/55842

File: 3d/particles/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# 3D Anti-Aliasing
```
Comment (jtnicholl) [inline 3d/particles/README.md:1]: Got a copy/paste error here.

**Comment by Calinou** [comment]
> In the future (in another PR) we could consider adding:
>
> - SDF mesh collisions
> - Heightfield collisions

I've added SDF and heightfield collision examples :slightly_smiling_face:

**Comment by clayjohn** [comment]
> > In the future (in another PR) we could consider adding:
> > 
> > * SDF mesh collisions
> > * Heightfield collisions
> 
> I've added SDF and heightfield collision examples slightly_smiling_face

Looks amazing!

**Review by adamscott — APPROVED** [review]
Nice demo! Works flawlessly.

**Comment by Calinou** [comment]
Rebased and tested again on 4.0.beta11, it works as expected.

**Comment by Sidremus** [comment]
Has the particles demo been completely removed? I'm still looking for an example implementation of 3D trails.

**Comment by Calinou** [comment]
> Has the particles demo been completely removed? I'm still looking for an example implementation of 3D trails.

The particles demo is in the `4.0-dev` branch, and has no 3.x equivalent in `master`. Trails are only available in 4.0.beta, not 3.5.1.

**Pull request merged**